app/src/main/java/org/simple/clinic/drugs/selectionv2/PrescribedDrugEvents.kt

```diff
@@ -7,3 +7,5 @@ import java.util.UUID
 data class ProtocolDrugSelected(val drugName: String, val prescribedDrug: PrescribedDrug?) : UiEvent
```

Comment:
I spent a lot of time staring at `val prescribedDrug: PrescribedDrug?` because something felt wrong about its nullable nature and I realized it's because of the name!

Because the event is named as `ProtocolDrugSelected`, it looks wrong that the `drug` is nullable. How can a drug be selected if it's null?

I then realized that this param is the previously selected drug. Let's improve this name? Something along the lines of `previouslySelectedDosage`.

Reply:
The event is `ProtocolDrugClicked`(after renaming) which refers to clicking a _protocol_ drug and the argument being passed is a `prescribedDrug` which is a _prescription_ and not the protocol drug itself. 
The protocol drug cannot be `null` but a prescription for a protocol drug can be. This relationship is already established in our app. 
Also, its not a dosage that we are passing but the prescription itself. Hence naming it on the lines of dosage can be misleading. 
Does this make sense?

Reply:
Whoops sorry, I meant prescription and not dosage. Got a bit confused here. `forProtocolDrug` suffix isn't necessary because that's implied by the event name. I am trying to make it easy to understand why the item is nullable through the name itself. Let's just go with `prescription`. I think the larger problem here is the name `PrescribedDrug`. It should be renamed to `Prescription`.